devlib/module/cpufreq.py

```diff
@@ -95,6 +97,37 @@ def set_governor(self, cpu, governor, **kwargs):
         self.target.write_value(sysfile, governor)
         self.set_governor_tunables(cpu, governor, **kwargs)
 
+    @contextmanager
+    def use_governor(self, governor, cpus=None, **kwargs):
+        """
+        Use a given governor, then restore previous governor(s)
+
+        :param governor: Governor to use on all targeted CPUs (see :meth:`set_governor`)
+        :type governor: str
+
+        :param cpus: CPUs affected by the governor change (all by default)
+        :type cpus: list
+
+        :Keyword Arguments: Governor tunables, See :meth:`set_governor_tunables`
+        """
+        if not cpus:
+            cpus = range(self.target.number_of_cpus)
+
+        # Setting a governor & tunables for a cpu will set them for all cpus
+        # in the same clock domain, so only manipulating one cpu per domain
+        # is enough
+        domains = set(self.get_affected_cpus(cpu)[0] for cpu in cpus)
+        prev_governors = {cpu : (self.get_governor(cpu), self.get_governor_tunables(cpu))
+                          for cpu in domains}
+
+        for cpu in domains:
+            self.set_governor(cpu, governor, **kwargs)
+
+        yield
```

Comment:
Should surround this in a  try/finally block to ensure the governor is restored (or at least an attempt is made to restore it) even if an exception occurs during context execution.

Reply:
For some reason I wrote this thinking that was done implicitly with context managers... Let me fix my code & my thoughts